plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -3804,36 +3806,6 @@ private Optional<CatalogSchemaTableName> redirectTableToHudi(ConnectorSession se
         return Optional.empty();
     }
 
-    private static TableNameSplitResult splitTableName(String tableName)
-    {
-        int metadataMarkerIndex = tableName.lastIndexOf('$');
-        return metadataMarkerIndex <= 0 ? // marker not found or at the beginning of tableName
-                new TableNameSplitResult(tableName, Optional.empty()) :
-                new TableNameSplitResult(tableName.substring(0, metadataMarkerIndex), Optional.of(tableName.substring(metadataMarkerIndex)));
-    }
-
-    private static class TableNameSplitResult
```

Comment:
"Remove TableNameSplitResult" looks unrelated. if it indeed is, let's move to separate PR.